contracts/SBCWrapper.sol

```diff
@@ -26,14 +27,16 @@ contract SBCWrapper is IERC677Receiver, PausableEIP1967Admin, Claimable, Reentra
     mapping(address => uint256) public tokenRate;
 
     SBCToken public immutable sbcToken;
+    SBCDepositContract public immutable sbcDepositContract;
 
     event Swap(address indexed token, address indexed user, uint256 amount, uint256 received);
     event SwapRateUpdated(address indexed token, uint256 rate);
     event TokenSwapEnabled(address indexed token);
     event TokenSwapPaused(address indexed token);
 
-    constructor(SBCToken _sbcToken) {
+    constructor(SBCToken _sbcToken, SBCDepositContract _depositContract) {
         sbcToken = _sbcToken;
+        sbcDepositContract = _depositContract;
```

Comment:
So, is it safe to deploy the new wrapper implementation manually and use it later through deployed proxy: both the token address and the deposit contract address will be initialized properly as part of the implementation deployment?

Reply:
Yes, everything should work smoothly, since the new variable is not stored in the contract storage. Instead, it is embedded into the deployed bytecode directly.